src/pq.py

```diff
@@ -0,0 +1,48 @@
+"""Implement a priority queue."""
+
+
+from math import inf
+
+
+class PikaQ(object):
+    """The Priority Queue Class."""
+
+    def __init__(self, vals=[]):
+        """Initialize a priority que."""
+        self.pq = {}
+        if hasattr(vals, "__iter__"):
+            try:
+                for val, pri in vals:
+                    self.insert(val, pri)
+            except TypeError:
+                for val in vals:
+                    self.insert(val)
+            except ValueError:
+                print("Too many values inserted in one of your lists")
+        elif not hasattr(type(vals, "__iter__")):
+            self.insert(vals)
+        else:
+            raise ValueError("Need to submit either a single value, a list of values, or a list of values tupled/listed with priorities.")
+
+    def insert(self, item, pri=float(-inf)):
```

Comment:
after you remove the import statement, you will see that this is not quite correct